ksql-execution/src/main/java/io/confluent/ksql/execution/plan/StreamGroupByKey.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.plan;
+
+import com.google.errorprone.annotations.Immutable;
+import io.confluent.ksql.execution.builder.KsqlQueryBuilder;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+@Immutable
+public class StreamGroupByKey<S, G> implements ExecutionStep<G> {
```

Comment:
These were originally meant to be the full stream and table types. I've switched it to just be key types now (in this PR for just the grouped steps, but I'll do that for all the steps). So this will go away.